tests/bsim/bluetooth/host/att/mtu_update/src/main.c

```diff
@@ -39,59 +41,84 @@ extern void run_peripheral_sample(uint8_t *notify_data, size_t notify_data_size,
 
 extern enum bst_result_t bst_result;
 
-#define CREATE_FLAG(flag) static atomic_t flag = (atomic_t)false
-#define SET_FLAG(flag) (void)atomic_set(&flag, (atomic_t)true)
-#define WAIT_FOR_FLAG(flag)                                                                        \
-	while (!(bool)atomic_get(&flag)) {                                                         \
-		(void)k_sleep(K_MSEC(1));                                                          \
-	}
-
 #define WAIT_TIME (20e6) /* 20 seconds */
 #define PERIPHERAL_NOTIFY_TIME ((WAIT_TIME - 10e6) / 1e6)
 
-CREATE_FLAG(flag_notification_received);
-uint8_t notify_data[100] = {};
-uint8_t is_data_equal;
-
-static uint8_t notify_cb(struct bt_conn *conn, struct bt_gatt_subscribe_params *params,
-			 const void *data, uint16_t length)
-{
-	printk("BSIM NOTIFY_CALLBACK\n");
-
-	is_data_equal = (length == sizeof(notify_data) && !memcmp(notify_data, data, length));
+#define CREATE_FLAG(flag) static atomic_t flag = 0
+#define INC_FLAG(flag) (void)atomic_inc(&flag)
+#define GET_FLAG(flag) atomic_get(&flag)
+#define WAIT_FOR_FLAG(flag)                                                                       \
+	while (atomic_get(&flag) == 0) {                                                          \
+		(void)k_sleep(K_MSEC(1));                                                         \
+	}
```

Comment:
I understand why you've change this, but I'm unsure if they should keep the `FLAG` part of the name. Flags are typically either set of unset (boolean), and not an integer value

You could consider just making `central_mtu_count` a `volatile size_t` or a semaphore instead, and leave the remaining flags as is.